bigtable/client/row_range_test.cc

```diff
@@ -169,3 +169,520 @@ TEST(RowRangeTest, ContainsEndingAt) {
+  using R = bigtable::RowRange;
+  EXPECT_EQ(R::RightOpen("a", "d"), R::RightOpen("a", "d"));
+  EXPECT_NE(R::RightOpen("a", "d"), R::RightOpen("a", "c"));
+  EXPECT_NE(R::RightOpen("a", "d"), R::RightOpen("b", "d"));
+  EXPECT_NE(R::RightOpen("a", "d"), R::LeftOpen("a", "d"));
+  EXPECT_NE(R::RightOpen("a", "d"), R::Closed("a", "d"));
+  EXPECT_NE(R::RightOpen("a", "d"), R::Open("a", "d"));
+}
+
+TEST(RowRangeTest, EqualsLeftOpen) {
+  using R = bigtable::RowRange;
+  EXPECT_EQ(R::LeftOpen("a", "d"), R::LeftOpen("a", "d"));
+  EXPECT_NE(R::LeftOpen("a", "d"), R::LeftOpen("a", "c"));
+  EXPECT_NE(R::LeftOpen("a", "d"), R::LeftOpen("b", "d"));
+  EXPECT_NE(R::LeftOpen("a", "d"), R::RightOpen("a", "d"));
+  EXPECT_NE(R::LeftOpen("a", "d"), R::Closed("a", "d"));
+  EXPECT_NE(R::LeftOpen("a", "d"), R::Open("a", "d"));
+}
+
+TEST(RowRangeTest, EqualsClosed) {
+  using R = bigtable::RowRange;
+  EXPECT_EQ(R::Closed("a", "d"), R::Closed("a", "d"));
+  EXPECT_NE(R::Closed("a", "d"), R::Closed("a", "c"));
+  EXPECT_NE(R::Closed("a", "d"), R::Closed("b", "d"));
+  EXPECT_NE(R::Closed("a", "d"), R::RightOpen("a", "d"));
+  EXPECT_NE(R::Closed("a", "d"), R::LeftOpen("a", "d"));
+  EXPECT_NE(R::Closed("a", "d"), R::Open("a", "d"));
+}
+
+TEST(RowRangeTest, EqualsOpen) {
+  using R = bigtable::RowRange;
+  EXPECT_EQ(R::Open("a", "d"), R::Open("a", "d"));
+  EXPECT_NE(R::Open("a", "d"), R::Open("a", "c"));
+  EXPECT_NE(R::Open("a", "d"), R::Open("b", "d"));
+  EXPECT_NE(R::Open("a", "d"), R::RightOpen("a", "d"));
+  EXPECT_NE(R::Open("a", "d"), R::LeftOpen("a", "d"));
+  EXPECT_NE(R::Open("a", "d"), R::Closed("a", "d"));
+}
+
+TEST(RowRangeTest, EqualsStartingAt) {
+  using R = bigtable::RowRange;
+  EXPECT_EQ(R::StartingAt("a"), R::StartingAt("a"));
+  EXPECT_NE(R::StartingAt("a"), R::StartingAt("b"));
+  EXPECT_NE(R::StartingAt("a"), R::RightOpen("a", "d"));
+  EXPECT_NE(R::StartingAt("a"), R::LeftOpen("a", "d"));
+  EXPECT_NE(R::StartingAt("a"), R::Open("a", "d"));
+  EXPECT_NE(R::StartingAt("a"), R::Closed("a", "d"));
+}
+
+TEST(RowRangeTest, EqualsEndingAt) {
+  using R = bigtable::RowRange;
+  EXPECT_EQ(R::EndingAt("b"), R::EndingAt("b"));
+  EXPECT_NE(R::EndingAt("b"), R::EndingAt("a"));
+  EXPECT_NE(R::EndingAt("b"), R::RightOpen("a", "b"));
+  EXPECT_NE(R::EndingAt("b"), R::LeftOpen("a", "b"));
+  EXPECT_NE(R::EndingAt("b"), R::Open("a", "b"));
+  EXPECT_NE(R::EndingAt("b"), R::Closed("a", "b"));
+}
+
+// This is a fairly exhausting (and maybe exhaustive) set of cases for
+// intersecting a RightOpen range against other ranges.
+using R = bigtable::RowRange;
+
+TEST(RowRangeTest, IntersectRightOpen_Empty) {
+  auto tuple = R::RightOpen("c", "m").Intersect(R::Empty());
+  EXPECT_FALSE(std::get<0>(tuple));
+}
+
+TEST(RowRangeTest, IntersectRightOpen_CompletelyBelow) {
+  auto tuple = R::RightOpen("c", "m").Intersect(R::RightOpen("a", "b"));
+  EXPECT_FALSE(std::get<0>(tuple));
+}
+
+TEST(RowRangeTest, IntersectRightOpen_MatchingBoundariesBelow) {
```

Comment:
Same comment about considering \0 in at least one test

Reply:
Done. At some point you are just testing that `std::string` works correctly.